worlds/lingo/items.py

```diff
@@ -0,0 +1,98 @@
+from typing import Dict, NamedTuple, Optional, List
+from BaseClasses import Item, MultiWorld
+from .Options import get_option_value
+from .static_logic import StaticLingoLogic
+
+
+class ItemData(NamedTuple):
+    """
+    ItemData for an item in Lingo
+    """
+    code: Optional[int]
+    progression: bool
+    trap: bool
+    mode: Optional[str]
+    event: bool
+    door_ids: List[str]
+    painting_ids: List[str]
+
+    def should_include(self, world: MultiWorld, player: int) -> bool:
+        if self.mode == "colors":
+            return get_option_value(world, player, "shuffle_colors") > 0
+        elif self.mode == "doors":
+            return get_option_value(world, player, "shuffle_doors") > 0
+        elif self.mode == "orange tower":
+            # door shuffle is on and tower isn't progressive
+            return get_option_value(world, player, "shuffle_doors") > 0\
+                and not get_option_value(world, player, "progressive_orange_tower")
+        elif self.mode == "complex door":
+            return get_option_value(world, player, "shuffle_doors") == 2  # complex doors
+        elif self.mode == "door group":
+            return get_option_value(world, player, "shuffle_doors") == 1  # simple doors
+        elif self.mode == "special":
+            return False
+        else:
+            return True
+
+
+class LingoItem(Item):
+    """
+    Item from the game Lingo
+    """
+    game: str = "Lingo"
+
+
+class StaticLingoItems:
+    """
+    Defines the items that can be included in a Lingo world
+    """
+
+    base_id: int = 0
+
+    ALL_ITEM_TABLE: Dict[str, ItemData] = {}
+
+    def create_item(self, name: str, event: bool, progression: bool, trap: bool, mode: Optional[str] = None,
+                    door_ids: Optional[List[str]] = None, painting_ids: Optional[List[str]] = None):
+        new_id = None if event is True else self.base_id + len(self.ALL_ITEM_TABLE)
```

Comment:
Why does it matter if IDs change between versions? It seems like unnecessary human toil, since it should be transparent to clients.

Reply:
You know what, this might be an outdated belief. I asked the core devs this question all the way back when, and one of them told me not to do it (and I didn't press for a reason). But thinking about it, at the time, there were some really good reasons not to do it, that are different now. Probably worth asking in #archipelago-dev if core devs still feel the same?

Reply:
In case you don't know the answer to this question the reason you want your IDs to be consistent is because your client is expected to be backwards compatible one version so people can play an 0.4.1 game and an 0.4.2 game without having to juggle mods. This isn't necessarily required but it isn't that much more effort to comply to and just reduces edge cases.